### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by Maclenn77**
Hi, @lu-jim :

I configured devise. It tooks a bit longer than expected due to some problems between versions. I tried simple_token_authentication and Doorkeeper gems, however those gems requires Actionpack version lower than 7 and we have the version 7. In consequence, I opted for devise-jwt, that didn't have any conflict with other versions.

I configured the gems following the documentation [here](https://github.com/waiting-for-dev/devise-jwt/wiki/Configuring-devise-for-APIs) and [here](https://github.com/waiting-for-dev/devise-jwt).

I also checked [this tutorial for JWT](https://www.youtube.com/watch?v=0ounptRMU9o).

As password authentication is not a mandatory requirement as I remember, I suggest to move on if we have some problem and then try to fix it if we have extra time.

Let me know if you found any problem.

Best regards,
JP

### CODE REVIEW

File: app/controllers/v1/sessions_controller.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class V1::SessionsController < ApplicationController
+  def create
+    user = User.where(email: params[:email]).first
+
+    if user&.valid_password?(params[:password])
+      render json: user.as_json(only: %i[id email], status: :created)
```
Comment (lu-jim) [inline app/controllers/v1/sessions_controller.rb]: At the moment, we are only rendering the user with email. Let's return the jwt authentication token so the user can add it to their requests.

File: db/seeds.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@
 #   movies = Movie.create([{ name: "Star Wars" }, { name: "Lord of the Rings" }])
 #   Character.create(name: "Luke", movie: movies.first)
 
-user1 = User.create!(username: 'Luma')
+user1 = User.create!(username: 'Luma', email: 'test@email.com', password: '12345678', password.confirmation: '12345678' )
```
Comment (lu-jim) [inline db/seeds.rb]: At the moment, the user1 seed is not properly being created. Let's change the `password.confirmation` to `password_confirmation`

**Review by lu-jim — CHANGES_REQUESTED** [review]
@Maclenn77 
Good job so far!
There are some issues that you still need to work on to go to the next feature but you are almost there!

Please check the comments for more information.

**Comment by Maclenn77** [comment]
Hi, @lu-jim :

I did several changes to the authentication tool to return a JWT token. However, there are some problems.

One of them is the devise version with Rails 7. 

There's a **DisabledSessionError when using authenticate_user! api only**, This is a bug reported and with a solution. However, the solution hasn't been added to the last version of Devise.

The problem appears when going to http://localhost:8000/member-data, a route created to test the API. So, maybe we can move on in that part.

When going to http://localhost:8000/users/sign_in, it appears a json file retrieving a message as it's sign_in 

{"message":"You are logged in."}

However, I haven't been able to do other actions even when the routes exists. I guess that it's required the route and the parameter in JSON. I was trying to use Postman to test that part, but I need help to use the tool. 

I was wondering that you weren't able to test the other version because in origins in CORS file, I only allowed Port:3000 in localhost. I change that part and now it allows any origin.

I hope that we can talk a bit in the afternoon about the project.

Greetings,
JP

File: app/controllers/users/registrations_controller.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    register_failed
+  end
+
+  def register_success
+    render json: { message: 'Signed up sucessfully.' }
+  end
+
+  def register_failed
+    render json: { message: 'Something went wrong.' }
```
Comment (lu-jim) [inline app/controllers/users/registrations_controller.rb:17]: I'm having issues with registration, and as we are not passing through errors, I'm not able to pinpoint the problem. 
image
## 
I used the same parameters on console and registration worked there.
## 
image


As this is not required at this stage, we can work on it later, but let me know if I'm missing anything, or if there is anything we can do to improve this.

File: config/initializers/cors.rb [hunk 4]
```diff
@@ -7,7 +7,7 @@
 
 Rails.application.config.middleware.insert_before 0, Rack::Cors do
   allow do
-    origins 'http://localhost:3000'
+    origins '*'
```
Comment (lu-jim) [inline config/initializers/cors.rb:10]: This was not the issue before, it was mostly that I expected to see the JWT token within the response. From what I see, the token is within the headers, so this time I was able to locate it.

**Review by lu-jim — APPROVED** [review]
Hello @Maclenn77,
Your feature is : 
### 🥇  Approved 🥇 
## Optional:
Please check the comments below so we can address this in the future.

**Pull request merged**